### PR TITLE
Support `sb run` on host directly without Docker

### PR DESCRIPTION
**Pull request by abuccts**
**Description**

Support `sb run` on host directly without Docker

**Major Revisions**
- Add `--no-docker` argument for `sb run`.
- Run on host directly if `--no-docker` if specified.
- Update docs and tests correspondingly.

### CODE REVIEW

**Review by cp5555 — APPROVED** [review]

File: docs/cli.md [hunk 1]
```diff
@@ -320,6 +322,12 @@ and default benchmarking configuration:
 sb run --docker-image superbench/cuda:11.1 --host-file ./host.ini
 ```
 
+Run kernel launch benchmarks on host directly without using Docker:
+```bash title="SB CLI"
+sb run --no-docker --host-list localhost --config-override \
+  superbench.enable=kernel-launch superbench.env.SB_MICRO_PATH=/path/to/superbenchmark
```
Comment (guoshzhao) [inline docs/cli.md:328]: `SB_MICRO_PATH=/path/to/superbenchmark` =>
`SB_MICRO_PATH=/path/to/superbenchmark_micro_bin`?

File: superbench/runner/playbooks/check_env.yaml [hunk 2]
```diff
@@ -22,6 +24,8 @@
     container: sb-workspace
     sb_nodes: '{{ hostvars.values() | map(attribute="ansible_hostname") | sort }}'
     sb_env: |
+      # sb env
+      HOST_WS={{ ansible_user_dir }}/sb-workspace
```
Comment (guoshzhao) [inline superbench/runner/playbooks/check_env.yaml:28]: What's the function of this line?

**Review by guoshzhao — APPROVED** [review]

File: setup.py [hunk 3]
```diff
@@ -166,6 +172,8 @@ def run(self):
             'pytest-cov>=2.11.1',
             'pytest-subtests>=0.4.0',
             'pytest>=6.2.2',
+            'types-markdown',
+            'types-pkg_resources',
```
[the remark this answers is not in the document]
Comment (abuccts) [reply setup.py:176]: it's just for type check in tests, not used in actual runs

File: docs/cli.md (hunk 1 above)
Comment (abuccts) [reply docs/cli.md:328]: the env is for build prefix not bin directory?
https://github.com/microsoft/superbenchmark/blob/6d895da83c5b0490bf26d7c4cdf2903abbb660e3/superbench/benchmarks/micro_benchmarks/micro_base.py#L152-L154

File: superbench/runner/playbooks/check_env.yaml (hunk 2 above)
Comment (abuccts) [reply superbench/runner/playbooks/check_env.yaml:28]: get the workspace path on the host (to change working directory when running)

**Pull request merged**